vllm/model_executor/models/interfaces.py

```diff
@@ -32,6 +34,38 @@ class SupportsMultiModal(Protocol):
     def __init__(self, *, multimodal_config: "MultiModalConfig") -> None:
         ...
 
+    def get_multimodal_embeddings(
+        self, **kwargs
+    ) -> Optional[Union["NestedTensors", List[Tuple["NestedTensors", str]]]]:
+        """
+        Returns multimodal embeddings generated from multimodal kwargs 
+        to be merged with text embeddings.
+        """
+        ...
+
+    # Only for models that support v0 chunked prefill
+    # TODO(ywang96): Remove this overload once v0 is deprecated
+    @overload
+    def get_input_embeddings(
+        self,
+        input_ids: torch.Tensor,
+        multimodal_embeddings: Optional["NestedTensors"] = None,
+        attn_metadata: Optional["AttentionMetadata"] = None,
+    ) -> torch.Tensor:
+        ...
+
+    def get_input_embeddings(
+        self,
+        input_ids: torch.Tensor,
+        multimodal_embeddings: Optional["NestedTensors"] = None,
+    ) -> torch.Tensor:
```

Comment:
```suggestion
    def get_multimodal_embeddings(
        self, **kwargs
    ) -> Optional[T]:
        """
        Returns multimodal embeddings generated from multimodal kwargs 
        to be merged with text embeddings.
        """
        ...

    # Only for models that support v0 chunked prefill
    # TODO(ywang96): Remove this overload once v0 is deprecated
    @overload
    def get_input_embeddings(
        self,
        input_ids: torch.Tensor,
        multimodal_embeddings: Optional[T] = None,
        attn_metadata: Optional["AttentionMetadata"] = None,
    ) -> torch.Tensor:
        ...

    def get_input_embeddings(
        self,
        input_ids: torch.Tensor,
        multimodal_embeddings: Optional[T] = None,
    ) -> torch.Tensor:
```

I think it's clean enough, as long as we be more transparent about our parameter types (like this)..

Reply:
Perhaps instead of a list of tuples, we can have a mapping from modalities to embeddings?

Reply:
I thought generic type `T` is a python 3.12 thing, isn't it?